examples/robust_vision_benchmark/cleverhans_attack_example/utils.py

```diff
@@ -0,0 +1,89 @@
+import cleverhans.model
+import tensorflow as tf
+import numpy as np
+
+
+def cleverhans_attack_wrapper(cleverhans_attack_fn, reset=True):
+    def attack(a):
+        session = tf.Session()
+        with session.as_default():
+            model = RVBCleverhansModel(a)
+            adversarial_image = cleverhans_attack_fn(model, session, a)
+            adversarial_image = np.squeeze(adversarial_image, axis=0)
+            if reset:
+                # optionally, reset to ignore other adversarials
+                # found during the search
+                a._reset()
+            # run predictions to make sure the returned adversarial
+            # is taken into account
+            min_, max_ = a.bounds()
+            adversarial_image = np.clip(adversarial_image, min_, max_)
+            a.predictions(adversarial_image)
+    return attack
+
+
+def py_func_grad(func, inp, Tout, stateful=True, name=None, grad=None):
+    """Custom py_func with gradient support
+
+    """
+    # Need to generate a unique name to avoid duplicates:
+    rnd_name = 'PyFuncGrad' + str(np.random.randint(0, 1E+8))
+
+    tf.RegisterGradient(rnd_name)(grad)
+    g = tf.get_default_graph()
+    with g.gradient_override_map({
+            "PyFunc": rnd_name,
+            "PyFuncStateless": rnd_name}):
+        return tf.py_func(func, inp, Tout, stateful=stateful, name=name)
+
+
+class RVBCleverhansModel(cleverhans.model.Model):
```

Comment:
I see one issue with this wrapper is that some attacks will try to use the probabilities (post-softmax) instead of the logits. In its current form, the method `get_logits()` will work but not `get_probs()`. Could you also provide access to the probabilities?

Reply:
We didn't implement get_probs because the Cleverhans model anyway falls back to softmax(logits) if probabilities are not defined, right?

Reply:
Not necessarily. If you take a look at the FastGradientMethod class, since you are passing it a `cleverhans.model.Model` object, it will call `model.get_probs()`: https://github.com/tensorflow/cleverhans/blob/master/cleverhans/attacks.py#L283

Reply:
But that should be fine since RBVCleverhansModel is derived from cleverhans.model.Model, and the latter implements get_prob such that it computes softmax(logits) if no layer with the name 'probs' exists: https://github.com/tensorflow/cleverhans/blob/master/cleverhans/model.py#L58-L60 Or am I missing something?

Reply:
That's correct, my bad I forgot about that default implementation we added